src/interactions/panZoomInteraction.ts

```diff
@@ -93,12 +93,14 @@ export module Interactions {
       var newCornerDistance = this._cornerDistance();
 
       if (this._xScale != null && newCornerDistance !== 0 && oldCornerDistance !== 0) {
-        PanZoom._magnifyScale(this._xScale, oldCornerDistance / newCornerDistance, oldCenterPoint.x);
-        PanZoom._translateScale(this._xScale, oldCenterPoint.x - newCenterPoint.x);
+        var invertedPX = this._xScale.invert(oldCenterPoint.x);
+        this._xScale.zoomLevel(this._xScale.zoomLevel() * oldCornerDistance / newCornerDistance);
+        PanZoom._translateScale(this._xScale, this._xScale.scale(invertedPX) - newCenterPoint.x);
```

Comment:
One of the implicit goals in this pull request as you see is to remove the `centerValue` idea from the scale magnification.  With this idea in mind, I needed to somehow remove this `oldCenterPoint.x` parameter.

To do this, first I need to magnify / affect the zoom level first.

Then, I need to translate to correct for the centerValue.

This translation is (you can see a similar calculation in the mousewheeling scenario)
`this._xScale.scale(invertedPX) - oldCenterPoint.x`

In short, the last two lines in this description should probably be...
```typescript
this._xScale.zoomLevel(this._xScale.zoomLevel() * oldCornerDistance / newCornerDistance);
PanZoom._translateScale(this._xScale, this._xScale.scale(invertedPX) - oldCenterPoint.x);
PanZoom._translateScale(this._xScale, oldCenterPoint.x - newCenterPoint.x);
```

However, since translation is a linear transform, I combined the last two translations into one, ending in the result you see here.